### PR TITLE
Separate profile for Iceberg REST and JDBC catalog tests

### LINKED ISSUE #15770: Iceberg tests sometimes get cancelled due to 1h timeout
Green case
* 37m https://github.com/trinodb/trino/actions/runs/3951547060/jobs/6766965334
* 57m https://github.com/trinodb/trino/actions/runs/3950441439/jobs/6765427757
* 42m https://github.com/trinodb/trino/actions/runs/3949120527/jobs/6761222486
* 46m https://github.com/trinodb/trino/actions/runs/3949003870/jobs/6760576407
* 38m https://github.com/trinodb/trino/actions/runs/3948486410/jobs/6759690966
* 43m https://github.com/trinodb/trino/actions/runs/3948269518/jobs/6758577284

Timeout case
* https://github.com/trinodb/trino/actions/runs/3951416192/jobs/6766275707
* https://github.com/trinodb/trino/actions/runs/3961830228/jobs/6788497074

### PR DESCRIPTION
**Pull request by ebyhr**
Fixes #15770

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -529,6 +529,8 @@ jobs:
             - { modules: plugin/trino-hive, profile: test-parquet }
             - { modules: plugin/trino-hudi }
             - { modules: plugin/trino-iceberg }
+            - { modules: plugin/trino-iceberg, profile: jdbc-tests }
+            - { modules: plugin/trino-iceberg, profile: rest-tests }
```
Comment (findepi) [inline .github/workflows/ci.yml:533]: Would it be OK to run JDBC and REST catalogs together (in one profile for simplicity, or in two profiles, but one CI task)?
Or, is each of them too big to have them together?

File: plugin/trino-iceberg/pom.xml [hunk 2]
```diff
@@ -569,6 +571,40 @@
...
+                                <include>**/io/trino/plugin/iceberg/catalog/jdbc/Test*.java</include>
+                            </includes>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+
+        <profile>
+            <id>rest-tests</id>
```
Comment (findepi) [inline plugin/trino-iceberg/pom.xml:592]: if you combine them into single profile, you can call it eg `additional-catalog-tests`

**Review by findepi — APPROVED** [review]
thank you!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ebyhr) [reply .github/workflows/ci.yml]: I suppose the single job is okay. Merged into one.

**Review by findepi — APPROVED** [review]

**Pull request merged**